packages/account/src/Sections/Assessment/FinancialAssessment/financial-assessment.tsx

```diff
@@ -267,10 +264,18 @@ const FinancialAssessment = observer(() => {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
-    const onSubmit = async (values: FormikValues, { setSubmitting, setStatus }: FormikHelpers<any>) => {
+    const onSubmit = async (
+        values: TFinancialInformationForm,
+        { setSubmitting, setStatus }: FormikHelpers<TFinancialInformationForm>
+    ) => {
         setStatus({ msg: '' });
         setIsBtnLoading(true);
-        const form_payload: any = {
+
+        if (shouldHideOccupationField(values?.employment_status || employment_status)) {
+            delete values?.occupation;
+        }
```

Comment:
Do we need this in onsubmit?
we are already hiding occupation field if shouldHideOccupationField is true, isnt?

Reply:
The field is hidden, but the prop exists in the form values. Hence before submission, the payload data is cleaned to remove unnecessary values